### PR TITLE
Update network for Medicalc

### PR DESCRIPTION
**Pull request by Finemas**
I edited requests according to apiary, which means to remove the document mode and add department model.

### CODE REVIEW

File: zScanner/DomainModel/DepartmentDomainModel.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+struct DepartmentDomainModel {
+    var id: String
+    var name: String
+}
+
+extension DepartmentDomainModel: Equatable {
+    static func == (lhs: DepartmentDomainModel, rhs: DepartmentDomainModel) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
```
Comment (skorepak) [inline zScanner/DomainModel/DepartmentDomainModel.swift]: You don't need to implement the `==` function. Just add Equatable directly to the first definition.. 
Or is there a reason why you only compare id instead of id and name?  

Where are you comparing it?

File: zScanner/DomainModel/DocumentTypeDomainModel.swift [hunk 2]
```diff
@@ -35,4 +22,4 @@ extension DocumentTypeDomainModel: ListItem {
 }
 
 // MARK: -
-typealias DocumentDict = [DocumentMode: [DocumentTypeDomainModel]]
+//typealias DocumentDict = [DocumentMode: [DocumentTypeDomainModel]]
```
Comment (skorepak) [inline zScanner/DomainModel/DocumentTypeDomainModel.swift]: Don't leave commented code behind. Remove it if it's unused

File: zScanner/Services/Storage/Model/DcumentTypeDatabaseModel.swift [hunk 3]
```diff
@@ -12,19 +12,16 @@ import RealmSwift
...
         self.name = documentType.name
-        self.mode = documentType.mode
     }
-
-    var mode: DocumentMode {
-        get { return DocumentMode(rawValue: _mode)! }
-        set { self._mode = newValue.rawValue }
+    
+    override class func primaryKey() -> String {
+        return "id"
```
Comment (skorepak) [inline zScanner/Services/Storage/Model/DcumentTypeDatabaseModel.swift:24]: Are you sure about this change? Why did you add primary key?

File: zScanner/Services/Tracking/Event.swift [hunk 4]
```diff
@@ -12,8 +12,6 @@ enum Event {
     case login
     case logout
     
-    case documentModeSelected(DocumentMode)
```
Comment (skorepak) [inline zScanner/Services/Tracking/Event.swift:15]: don't remove the tracking event, it can stay there unused :) 
More like add department elected event with Department id/code

File: zScanner/SupportFiles/Helpers/Config.swift [hunk 5]
```diff
@@ -10,7 +10,7 @@ import Foundation
 
 enum Config {
     static let productionURL: String = "https://tempra.ikem.seacat/api-zscanner/v3"
-    static let testingURL: String = "https://desolate-meadow-62603.herokuapp.com/api-zscanner/v3"
+    static let testingURL: String = "https://private-anon-7cd588d2e4-skorepak.apiary-mock.com/medicalc/v3.1"
```
Comment (skorepak) [inline zScanner/SupportFiles/Helpers/Config.swift]: don't remove testingUrl, add separated apiaryUrl and whole `.apiary` environment

File: zScanner/ViewModels/NewDocument/NewDocumentTypeViewModel.swift [hunk 6]
```diff
@@ -38,22 +36,16 @@ class NewDocumentTypeViewModel {
     }
     
     // MARK: Helpers
-    private func fields(for mode: DocumentMode) -> [FormField] {
+    private func getFields() -> [FormField] {
```
Comment (skorepak) [inline zScanner/ViewModels/NewDocument/NewDocumentTypeViewModel.swift]: getFields is not the best name. convert it into variable `fields: [FormField]` :) you don't need the parameter anymore, so it will be nicer

File: zScanner/DomainModel/DepartmentDomainModel.swift (hunk 1 above)
Comment (Finemas) [reply zScanner/DomainModel/DepartmentDomainModel.swift]: There is actually no reason to compare only by id. Thank you

File: zScanner/DomainModel/DocumentTypeDomainModel.swift (hunk 2 above)
Comment (Finemas) [reply zScanner/DomainModel/DocumentTypeDomainModel.swift]: I agree with you. I just wasn't sure how document mode infects the app flow. So I try to remove it carefully.

File: zScanner/Services/Storage/Model/DcumentTypeDatabaseModel.swift (hunk 3 above)
Comment (Finemas) [reply zScanner/Services/Storage/Model/DcumentTypeDatabaseModel.swift:24]: I'm not sure about it but it says that the DocumentTypeDatabaseModel needs to have a primary key and it doesn't work without it.

**Review by skorepak — APPROVED** [review]
